### PR TITLE
Task-37846 : Active DLP restore action

### PR DESCRIPTION
**Pull request by ahmadmido**

### CODE REVIEW

**Review by rdenarie — DISMISSED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java [hunk 1]
```diff
@@ -4,75 +4,94 @@
 import org.exoplatform.commons.dlp.domain.DlpPositiveItemEntity;
 import org.exoplatform.commons.dlp.dto.DlpPositiveItem;
 import org.exoplatform.commons.dlp.service.DlpPositiveItemService;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.ListenerService;
 import org.exoplatform.services.log.ExoLogger;
 import org.exoplatform.services.log.Log;
-import org.exoplatform.services.organization.OrganizationService;
 
+import java.awt.*;
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: Why do you need awt dependency ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java [hunk 2]
```diff
@@ -64,6 +64,21 @@ public void deleteDlpPositiveItem(Long itemId) {
...
     }
   }
+ 
+   @Override
+  public void restoreDlpPositiveItem(Long itemId) {
+    try {
+      DlpPositiveItemEntity dlpPositiveItemEntity = dlpPositiveItemDAO.find(itemId);
+      if (dlpPositiveItemEntity != null) {
+        dlpPositiveItemDAO.delete(dlpPositiveItemEntity);
+        listenerService.broadcast(new Event("dlp.listener.event.move.document", null, dlpPositiveItemEntity.getReference()));
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]:
```suggestion
        listenerService.broadcast(new Event("dlp.listener.event.restore.item", null, dlpPositiveItemEntity.getReference()));
```
I think we should rename the event

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public void deleteDlpPositiveItem(Long itemId) {
       long startTime = System.currentTimeMillis();
       dlpPositiveItemDAO.delete(dlpPositiveItemEntity);
       try {
-        listenerService.broadcast(new Event("dlp.listener.event.delete.document", null, dlpPositiveItemEntity.getReference()));
+        listenerService.broadcast(new Event("dlp.listener.event.delete.item", null, dlpPositiveItemEntity.getReference()));
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java:60]: did you change also in configuration ?

Comment (ahmadmido) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java:60]: yes in ECMS

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/RestoredDlpItemServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    try {
+      return fillRestoredDlpItemFromEntity(restoredDlpItemEntity);
+    }catch (Exception e){
+      LOG.warn("The restored Dlp Item's {} not found.", itemReference);
+      return null;
+    }
+  }
+
+  @Override
+  public List<RestoredDlpItem> getRestoredDlpItems(int offset, int limit) throws Exception {
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/RestoredDlpItemServiceImpl.java]: Why this one is necessary ? In which case should we get a list of RestoreDlpItems ?

File: commons-dlp/src/main/resources/db/changelog/exo-dlp.db.changelog-1.0.0.xml [hunk 5]
```diff
@@ -65,5 +65,16 @@
...
+
+  <changeSet author="exo-dlp" id="1.0.0-6">
+    <createTable tableName="RESTORED_DLP_ITEMS">
+      <column name="RESTORED_ITEM_ID" type="BIGINT" autoIncrement="${autoIncrement}" startWith="1">
+        <constraints nullable="false" primaryKey="true" primaryKeyName="PK_RESTORED_DLP_ITEMS"/>
+      </column>
+      <column name="RESTORED_ITEM_REFERENCE" type="VARCHAR(50)"/>
+      <column name="RESTORED_DETECTION_DATE" type="TIMESTAMP"/>
+    </createTable>
+    <createSequence sequenceName="SEQ_RESTORED_DLP_ITEMS_ID" startValue="1"/>
```
Comment (rdenarie) [inline commons-dlp/src/main/resources/db/changelog/exo-dlp.db.changelog-1.0.0.xml:80]: the createSequence should be inside another changetset, applicable only for psql

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**